File: app/src/main/java/org/senatov/compare/CompareResult.java
/*
 * CompareResult.java — result of a compare operation.
 * Carries left/right item lists, diff count, summary text.
 * Iakov Senatov, 2026
 */
package org.senatov.compare;

import org.senatov.model.CompareLineItem;

import java.util.List;


public record CompareResult(
        List<CompareLineItem> leftItems,
        List<CompareLineItem> rightItems,
        int diffCount,
        String summary
) {

    public boolean isIdentical() {
        return diffCount == 0;
    }


    public String statusText() {
        if (isIdentical()) {
            return "✅ identical";
        }
        return "≠ " + diffCount + " differences";
    }
}

File: app/src/main/kotlin/org/senatov/compare/CompareResult.kt
/*
 * CompareResult — result of compare op.
 * left/right item lists, diff count, summary.
 * Iakov Senatov, 2026
 */
package org.senatov.compare

import org.senatov.model.CompareLineItem


data class CompareResult(
    val items: PairedCompareItems,
    val diffCount: Int,
    val summary: String
) {
    val leftItems: List<CompareLineItem> get() = items.left
    val rightItems: List<CompareLineItem> get() = items.right

    val isIdentical: Boolean get() = diffCount == 0


    fun statusText(): String =
        if (isIdentical) "✅ identical"
        else "≠ $diffCount differences"
}
